source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -182,6 +182,12 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
     // Only send headers if the response is ok.
     if (response->status == CheckStatus::OK) {
       ENVOY_STREAM_LOG(trace, "ext_authz filter added header(s) to the request:", *callbacks_);
+      if (config_->clearRouteCache() &&
+          (!response->headers_to_add.empty() || !response->headers_to_append.empty())) {
```

Comment:
Sorry, for my own education, what is the relation between the existence of `headers_to_add` or `headers_to_append` with the decision of clearing route cache?

Reply:
Sure, being able to route a request based on authorization response metadata is the whole point of clearing the cache. If some particular request has no metadata to be added/appended, just leave the route cache how it is. At least that is how I understand it.